node/mo/utils/db.ts

```diff
@@ -0,0 +1,86 @@
+import { MongoClient, Collection, Db, Document } from "mongodb";
+import { MOLINKS_CONFIG } from "./config";
+
+/**
+ * @returns A new MongoClient connected with the default database URL.
+ *
+ * @note This will need to be closed!
+ */
+export async function newMongoClient(): Promise<MongoClient> {
+    return MongoClient.connect(MOLINKS_CONFIG.DATABASE_URL, {
+        appName: 'molinks',
+        connectTimeoutMS: 1_000,
+    });
+}
+
+/**
+ * Perform work with a new MongoClient, automatically closing it when the handler exits
+ *
+ * @param fn A handler that performs work with the given MongoClient
+ * @returns The return value from the callback
+ */
+export async function withClient<T>(fn: (client: MongoClient) => T): Promise<Awaited<T>> {
```

Comment:
I would expect the app to open a single client for the lifetime of the process and only close it before exiting the server. Do we need a withClient helper in that case?

Reply:
That's possible, but I wouldn't know the best way to do that within NextJS, as it would need to install the client in some way that is accessible to request handlers but will definitely be closed during shutdown/idle.